src/Overlay/GlobalOverlay/GlobalOverlay.tsx

```diff
@@ -38,6 +39,12 @@ export const GlobalOverlay: FC<GlobalOverlayProps> = ({
     onEscape: () => closeOnEscape && onClose?.()
   });
 
+  useEffect(() => {
+    if (children !== undefined) {
```

Comment:
Should we have a check for `open` here?

Reply:
Yes. I must have missed it because I did the disable-call on the ref for the OverlayPortal before. But the children seem to be the thing to call this on.